From 7d9a4dd8ed3546f5d16c46e8b0a13d39ecfb82ee Mon Sep 17 00:00:00 2001
From: qiongz <qiongz@codeaurora.org>
Date: Wed, 14 Jun 2017 15:42:56 +0800
Subject: [PATCH 07/10] Fix imsi info error when insert some china operator
 card.

Imsi only show mccmnc when the phone type is cdma phone since the
length of return from RIL is 0 but not NULL for mMin. if mMin is
NULL or length is 0, mPhone.getSubscriberId() will get NULL when
the phone type is cdma, hence the flow of get imsi will move on
to retrieve the imsi from IccRecords.

Change-Id: Idda892c736db046d8924bb45a5fbb2a5f2a6d248
CRs-Fixed: 2212375
---
 .../com/android/internal/telephony/ServiceStateTracker.java     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/ServiceStateTracker.java b/src/java/com/android/internal/telephony/ServiceStateTracker.java
index b3a3bd14b..f4611eb5b 100755
--- a/src/java/com/android/internal/telephony/ServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/ServiceStateTracker.java
@@ -1778,7 +1778,7 @@ public class ServiceStateTracker extends Handler {
                 .getSystemService(Context.TELEPHONY_SERVICE))
                 .getSimOperatorNumericForPhone(mPhone.getPhoneId());
 
-        if (!TextUtils.isEmpty(operatorNumeric) && getCdmaMin() != null) {
+        if (!TextUtils.isEmpty(operatorNumeric) && !TextUtils.isEmpty(getCdmaMin())) {
             return (operatorNumeric + getCdmaMin());
         } else {
             return null;
-- 
2.31.1.606.gdf6c4f722c

